lib/sagas/keychain.js

```diff
@@ -180,8 +190,28 @@ export function * createIdentityKeyPair () {
   }
 }
 
-function * keychainSaga () {
+export function * canSignFor (address) {
+  const settings = yield select(networkSettingsForAddress, address)
+  if (typeof settings.hdindex === 'number') {
+    const idIndex = yield select(hdIdentityIndex, address)
+    const actIndex = yield select(hdAccountIndex, address)
+    const r = yield call(addressFor, idIndex, actIndex)
+    return r.address === settings.deviceAddress
+  } else {
+    const addresses = yield call(NativeModules.NativeSignerModule.allAddresses)
```

Comment:
This is returning empty list on Android. Because of that (?) after reloading freshly installed app, you get Migration screen. When I changed this line to 
```
const addresses = yield call(listSeedAddresses)
```
everything seems to be working fine, but I'm not sure if that is not breaking something else...